pkg/commands/catalog/mirror/mirror.go

```diff
@@ -104,6 +114,52 @@ func Mirror(options Options) error {
 		}
 	}
 
+	if options.Download {
+		// Create a temporary directory to place the oci-archive files
+		ociArchiveDir, err := os.MkdirTemp("", "oci-archive")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(ociArchiveDir)
+		counter := 1
+		for _, image := range images {
+			imageInfo, err := imageUtil.SplitImage(image)
+			if err != nil {
+				return err
+			}
+			log.Debugf("Copying %s:%s to system", imageInfo.BaseImage, imageInfo.Tag)
+			timeoutHappened := true
+			for i := 0; i < 5; i++ {
```

Comment:
cleaner to a function copyOneImage so that you can directly return from loop if success and get rid of timeout flag altogether